app/tests/system/setup/setup.js

```diff
@@ -0,0 +1,142 @@
+/*
+ * Loop through the custom contexts that exist when the app starts
+ * and grab their names.
+ *
+ * To change the environment we're testing against, specify it in site.json.
+ * This should align with what the app is built against, which is declared in
+ * app/environment.js hostUrl
+ */
+
+var Site = require('../site.json');
+
+var Setup = {};
+
+var selectors = {
+    previewButton: '#authorize',
+    registeredAuctionList: '.js-auction-lists'
+};
+
+var expected = {
+    //EXAMPLE Webviews constructed for project.
+    'noInternet': /no-internet\.html/i,
+    'couldNotLoad': /could-not-load\.html/i
+};
+
+Setup.appPreview = function(browser) {
+    browser.contexts(function(result) {
+        browser
+            .session(function(type) {
+                browser
+                    .log('Running on platform - ' + type.value.platformName);
+                if (type.value.platformName === 'Android') {
+                    var contextsAndroid = result.value;
+                    browser
+                        .log('Starting Preview Process')
+                        .currentContext(function(ctx) {
+                            browser.log('The current context is ' + ctx.value);
+                        })
+                        .setContext(contextsAndroid[1])
+                        .useCss()
+                        .waitForElementVisible(selectors.previewButton)
+
+                    // Checking window handle
+                    .windowHandle(function(h) {
+                            browser.log('Window handle before clicking preview ' + h.value);
+                        })
+                        .click(selectors.previewButton)
+                        //This is only needed if using an adaptive build
+                        .pause(3000)
+                        .waitUntilMobified();
+                } else if (type.value.platformName === 'iOS') {
+                    browser.log('these are the contexts ' + result.value);
+                    var contextsiOS = result.value.slice(2);
+                    browser
+                        .log('Starting Preview Process iOS')
+                        .currentContext(function(ctx) {
+                            browser.log('The current context is ' + ctx.value);
+                        })
+                        .setContext(contextsiOS[contextsiOS.length - 1])
```

Comment:
This part is a little bit tricky, because in RBA project different tests might have different way to call setContext. So it probably needs 4 hours at least to have a try in RBA project to make sure it really works properly.